### ISSUE TITLE
Bug: turf/difference fails on a shared edge

### ISSUE THREAD

**Issue by PizzaBrandon** [Turfjs/turf#1393]
It appears difference is failing due to a bug in a supporting library: https://github.com/w8r/martinez/issues/35

Here is a runnable test that exhibits the behavior
```javascript
const difference = require('@turf/difference');
tape('Difference failure', (t) => {
  const polygon1 = {
    'type': 'Polygon',
    'coordinates': [
      [[ 10, 10 ], [ 80, 10 ], [ 80, 80 ], [ 10, 80 ], [ 10, 10 ]]
    ]
  };

  const polygon2 = {
    'type': 'Polygon',
    'coordinates': [
      [[ 20, 10 ], [ 30, 10 ], [ 30, 80 ], [ 20, 80 ], [ 20, 10 ]]
    ]
  };

  const result = difference(polygon1, polygon2);
});
// Expected result to be a multi-polygon
// Actual fails with exception `Each LinearRing of a Polygon must have 4 or more Positions.`
```
This failure happens since `martinez` returns a malformed polygon when two edges overlap. It returns a "good" difference and as well as a two- or three-point polygon. Could this be detected and handled prior to handing off to `helper.polygon`, which is where the exception occurs?

(Hoping creating a tracking issue between the two libraries helps find a common solution faster.)

**Comment by PizzaBrandon** [comment]
Additional note, there is a larger bug in `martinez` where it returns an incorrect result for the above test, but at least need to make sure `difference` doesn't exception.

**Comment by w8r** [comment]
linking the issue here w8r/martinez#81

**Comment by HansonYip** [comment]
I also found a similar problem about shared edge. 
```javascript
var p1 = [[[0,0],[2,2],[4,2],[2,0],[0,0]]];
var p2 = [[[1,0],[2,1],[3,1],[2,0],[1,0]]];
var differenced = difference(p1, p2);
```
But the result is 
```javascript
[[[[0,0],[1,0],[2,0],[2,1],[3,1],[4,2],[2,2],[0,0]]]]
```

**Comment by rowanwins** [comment]
Resolved in v6.2.0 onwards
